src/FragmentRegistry/ContentElement/DelegatingContentElementRenderer.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\FragmentRegistry\ContentElement;
+
+use Contao\ContentModel;
+use Contao\CoreBundle\ContaoCoreBundle;
+
+/**
+ * Class DelegatingContentElementRenderer.
+ *
+ * @author Yanick Witschi <https://github.com/toflar>
+ */
+class DelegatingContentElementRenderer implements ContentElementRendererInterface
+{
+    /**
+     * @var ContentElementRendererInterface[]
+     */
+    private $renderers;
+
+    /**
+     * ChainFrontendModuleRenderer constructor.
+     *
+     * @param ContentElementRendererInterface[] $renderers
+     */
+    public function __construct(array $renderers)
```

Comment:
Delegaters usually have an add methods. See https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Templating/DelegatingEngine.php#L34 for example.

Also, most cases I've seen use setter injection and not constructor injection, though the templating example does not…

Reply:
Contrary to the current implementation, it would (automatically) validate the array only contains classes implementing the correct interface…

Reply:
If it's not in the interface, I'm fine with that. Added in 537ba59d50d14cdb41621d920efcbe632ff87932.